lib/helpers/backend/makeAxiosRequest.ts

```diff
@@ -0,0 +1,34 @@
+import { AxiosRequestConfig } from "axios";
+import { StatusCodes } from "http-status-codes";
+import axios from "axios";
+
+export async function makeAxiosRequest(
+  endpoint: string,
+  options?: AxiosRequestConfig,
+  milliInterval?: number,
+  maxRetry?: number
+) {
+  maxRetry = maxRetry || 5;
+  milliInterval = milliInterval || 1000;
+  options = options || {};
+  async function getData(maxRetry: number) {
+    return await axios
+      .get(endpoint, options)
+      .then(function (response) {
+        return response.data;
+      })
+      .catch(function (error) {
+        if (
+          (maxRetry !== 1 &&
+            error.response.status >= StatusCodes.INTERNAL_SERVER_ERROR) ||
+          error?.response?.status === StatusCodes.TOO_MANY_REQUESTS
+        ) {
+          setTimeout(function () {
+            getData(--maxRetry);
```

Comment:
I don't think that this works, since the result of `getData` is not returned. I think you should be able to make the `.catch` function `async`, then return `getData`. In order to add a delay before getting, you could await a promise that returns after `milliInterval` (which you could implement with `setTimeout`)

@calebmech is there a better way to implement this?

Reply:
Filed removed. Similar fill will be included with book-scrapper update.

Will look for ways to improve it and ask Caleb about it. Thanks!